components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -103,17 +76,18 @@ export class DesktopDetailComponent implements OnInit, OnDestroy {
     this.downloadDropdownVisible = false;
   }
 
-  addCheckInLabels(checkInHistory: DailyNodeRunsStatus[]): DailyNodeRunsStatus[] {
-    let numWeeks = Math.floor(checkInHistory.length / 7);
-    checkInHistory.forEach((history: DailyNodeRunsStatus, index: number) => {
+  get labeledCheckInHistory() {
+    const buckets = this.checkInHistory.durations.buckets;
+    let numWeeks = Math.floor(buckets.length / 7);
+    return buckets.map((history: DailyNodeRunsStatus, index: number) => {
       const isStartOfWeek = (index % 7 === 0) && numWeeks > 0;
       const startOfWeekLabelText = numWeeks > 1 ? `${numWeeks} weeks ago` : `${numWeeks} week ago`;
```

Comment:
Account for zero here, also:
```suggestion
      const startOfWeekLabelText = numWeeks != 1 ? `${numWeeks} weeks ago` : `${numWeeks} week ago`;
```
Alternately, if you instantiate a `PluralizePipe`, you can do this:
```suggestion
      const startOfWeekLabelText = `${pluralizePipe.transform(numWeeks, 'week', '+s')} ago`;
```